Add logging framework.

kyler-m opened the pull request.

Commit:
Add logging framework.

Commit:
Cleanup/add README for logging.

pelmers commented on config/dn-log-conf.conf:
```diff
@@ -0,0 +1,17 @@
+* GLOBAL:
+   FORMAT               =  "%datetime{%Y-%M-%d %H:%m:%s.%g} [%level] %msg"
+   FILENAME             =  "logs/nn-log.log"
```
should it log to same file as namenode? maybe you mean dn-log.log

pelmers commented on rice-namenode/main.cc:
```diff
@@ -1,12 +1,24 @@
 using namespace client_namenode_translator;
 
 int main(int argc, char* argv[]) {
+	el::Configurations conf(LOG_CONFIG_FILE);
```
inconsistent tabs/space here

Commit:
Tweak README.

Commit:
Tweak README.

Tweak readme again.

Tweak readme again.

Tweak readme again.

Tweak readme again.

Commit:
Tweak readme again.

Commit:
Tweak readme again.

pelmers reviewed (commented):
Top-level remarks:
1. remove utility/logging/.ds_store
2. could replace the prints in the ERROR_AND_RETURN and ERROR_AND_FALSE macros with logs to error? (if there is no error log it would be nice to have one, but going to same file as other output is also okay)

Otherwise I approve. r+ with those changes

Commit:
Tweak readme again.

Commit:
Tweak readme again.

Commit:
Merge branch 'logging' of https://github.com/Rice-Comp413-2016/Rice-HDFS into logging

Commit:
Use LOG macro instead of cerr for ERROR macros.

pelmers commented:
Don't do these if it takes more than 5 minutes, but I would consider squashing all the readme tweaks into one commit (`rebase -i develop`, and put `f` in front of all tweak commits after the first).

And changing vagrant@vagrant the author field of commits to yourself (`commit --amend --author "first last x@y.com --no-edit"`).

pelmers commented:
adding to above, the rebase would remove the no-op self-merge commit

kyler-m commented:
I was planning on just taking advantage of Github's merge-and-squash to condense all of my commits into one

pelmers commented:
oh, perfect -- disregard
